core/base/document.php

```diff
@@ -1375,4 +1390,59 @@ private function add_handle_revisions_changed_filter() {
 	private function remove_handle_revisions_changed_filter() {
 		remove_filter( 'wp_save_post_revision_post_has_changed', [ $this, 'handle_revisions_changed' ] );
 	}
+
+	/**
+	 * Validate elements settings.
+	 *
+	 * Iterate all elements via the `validate_controls` method.
+	 *
+	 * @param array $elements
+	 *
+	 * @return array
+	 */
+	private function validate_elements_settings( array $elements ) {
+		return Plugin::$instance->db->iterate_data( $elements, function ( $element_data ) {
+			$instance = Plugin::$instance->elements_manager->create_element_instance( $element_data );
+
+			$element_data['settings'] = $this->validate_controls( $element_data['settings'], $instance->get_controls() );
+
+			return $element_data;
+		} );
+	}
+
+	/**
+	 * Validate settings of an element or a document.
+	 *
+	 * Avoid change settings like `html_tag` & `title_size` to a `script` tag.
+	 * Go over all Select & Select2 controls in order to handle also 3rd party settings.
+	 *
+	 * @param array $settings The setting to validate.
+	 * @param array $controls The controls config for the settings set.
+	 *
+	 * @return array
+	 */
+	private function validate_controls( array $settings, array $controls ) {
```

Comment:
It's better to add a method inside the control class for validating saved values